tests/spark_autologging/test_pyspark_ml_autologging.py

```diff
@@ -0,0 +1,237 @@
+
+
+def get_expected_class_tags(estimator):
+    return {
+        "estimator_name": estimator.__class__.__name__,
+        "estimator_class": estimator.__class__.__module__ + "." + estimator.__class__.__name__,
+    }
+
+
+def get_run_data(run_id):
+    client = mlflow.tracking.MlflowClient()
+    data = client.get_run(run_id).data
+    # Ignore tags mlflow logs by default (e.g. "mlflow.user")
+    tags = {k: v for k, v in data.tags.items() if not k.startswith("mlflow.")}
+    artifacts = [f.path for f in client.list_artifacts(run_id)]
+
+    RunData = namedtuple("RunData", ["params", "metrics", "tags", "artifacts"])
+    return RunData(data.params, data.metrics, tags, artifacts)
+
+
+def load_model_by_run_id(run_id):
+    return mlflow.spark.load_model("runs:/{}/{}".format(run_id, MODEL_DIR))
+
+
+def get_dataset_binomial(spark_session):
+    return spark_session.createDataFrame(
+        [(1.0, Vectors.dense(1.0)), (0.0, Vectors.sparse(1, [], []))] * 100, ["label", "features"]
+    )
+
+
+def test_basic_estimator(dataset_binomial):
+    mlflow.pyspark.ml.autolog()
+    lr = LinearRegression()
+    with mlflow.start_run() as run:
+        lr_model = lr.fit(dataset_binomial)
+    run_id = run.info.run_id
+    run_data = get_run_data(run_id)
+    assert run_data.params == truncate_param_dict(
+        stringify_dict_values(_get_instance_param_map(lr))
+    )
+    assert run_data.tags == get_expected_class_tags(lr)
+    assert MODEL_DIR in run_data.artifacts
+    loaded_model = load_model_by_run_id(run_id)
+    assert loaded_model.stages[0].uid == lr_model.uid
+
+
+@pytest.mark.skipif(
+    LooseVersion(pyspark.__version__) < LooseVersion("3.1"),
+    reason="This test require spark version >= 3.1",
+)
+def test_models_in_allowlist_exist(spark_session):  # pylint: disable=unused-argument
+    mlflow.pyspark.ml.autolog()  # initialize the variable `mlflow.pyspark.ml._log_model_allowlist`
+
+    def model_does_not_exist(model_class):
+        module_name, class_name = model_class.rsplit(".", 1)
+        try:
+            module = importlib.import_module(module_name)
+            return not hasattr(module, class_name)
+        except ModuleNotFoundError:
+            return True
+
+    non_existent_classes = list(
+        filter(model_does_not_exist, mlflow.pyspark.ml._log_model_allowlist)
+    )
+    assert len(non_existent_classes) == 0, "{} in log_model_allowlist don't exist".format(
+        non_existent_classes
+    )
+
+
+def test_autolog_does_not_terminate_active_run(dataset_binomial):
+    mlflow.pyspark.ml.autolog()
+    mlflow.start_run()
+    lr = LinearRegression()
+    lr.fit(dataset_binomial)
+    assert mlflow.active_run() is not None
+    mlflow.end_run()
+
+
+def test_meta_estimator_fit_performs_logging_only_once(dataset_binomial):
+    from pyspark.ml.classification import LinearSVC, OneVsRest
+
+    mlflow.pyspark.ml.autolog()
+    with mock.patch("mlflow.log_params") as mock_log_params, mock.patch(
```

Comment:
Why use mocks here? Could we just fit the one vs rest model and verify that there's a single run with the desired params, tags, and model?